api/src/main/java/org/openmrs/module/registrationcore/api/mpi/pixpdq/PdqPatientFetcher.java

```diff
@@ -22,17 +26,26 @@
     @Qualifier("registrationcore.Hl7SenderHolder")
     private Hl7SenderHolder hl7SenderHolder;
 
+    @Autowired
+    private MpiProperties mpiProperties;
+
     protected final Log log = LogFactory.getLog(this.getClass());
 
     @Override
     public Patient fetchMpiPatient(String patientId) {
-        PatientIdentifier identifier = new PatientIdentifier(patientId, null, null);
-        Map<String, String> queryParams = new HashMap<String, String>();
-
-        queryParams.put("@PID.3.1", identifier.getIdentifier());
+        Patient patient = Context.getPatientService().getPatient(Integer.valueOf(patientId));
+        if (patient == null) {
+            patient = Context.getPatientService().getPatientByUuid(patientId);
+        }
```

Comment:
there is a problem with this is method - if patientID is a UUID, we will get an exception on line 36. PatientId should mean only one thing here, if that's too broad, we can consider changing the method signature.